### DISCUSSION TITLE
Feature request: Add `true_values` and `false_values` for CSV when reading BOOLEAN fields.

### DISCUSSION THREAD

**Discussion by fdcastel** [duckdb/duckdb#15417] [General]
Add `true_values` and `false_values` parameters for CSV when reading BOOLEAN fields.

https://duckdb.org/docs/data/csv/overview#parameters

It should accept a list containing all possible values of a BOOLEAN column, similar to the `nullstr` parameter. E.g.

```sql
    read_csv(...,
             header = false,
             columns = {'first': 'UINTEGER',
                        'second': 'BOOLEAN'},
             nullstr = ['', '0'],
             true_values = ['Y', '1', 'S', 'J'],
             false_values = ['N', '0'],
             delim = ';')
```

Rationale: CSV is not a strictly followed standard (to say the least), and it often varies significantly around regions. 

Like date format, or decimal separator, non-English speakers often use the first letter of their own equivalents for "Yes" and "No" for columns of this type.

**Comment by pdet** [comment]
Thanks for opening the discussion.

Since [Pandas supports](https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html) this. I'm not opposed to adding these options.

With that said I'm also not a fan of expanding the csv options too much, I do wonder, if expanding the cast rules of `VARCHAR` -> `BOOL` would suffice.
